ios/browser/ui/webui/brave_url_data_source_ios.mm

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/ui/webui/brave_url_data_source_ios.h"
+
+#include "base/no_destructor.h"
+#include "base/strings/strcat.h"
+#include "ios/components/webui/web_ui_url_constants.h"
+
+namespace {
+
+// A chrome-untrusted data source's name starts with chrome-untrusted://.
+bool IsChromeUntrustedDataSource(const web::URLDataSourceIOS* source) {
+  static const base::NoDestructor<std::string> kChromeUntrustedSourceNamePrefix(
```

Comment:
`GURL(source->GetSource()).SchemeIs(kChromeUIUntrustedScheme)`

Reply:
you probably don't need this helper method with the above, but if you keep it, I would pass in the string and not the `URLDataSourceIOS`.

Reply:
Fixed.